website/content/docs/use-cases/hcp-hashistack-quickstart.mdx

```diff
@@ -0,0 +1,495 @@
+We will deploy a custom HashiCorp version of the popular game "2048".
+Please clone the repo if you'd like to follow along the tutorial on your local machine.
+
+## Create service principal and key
+
+To leverage the Terraform integration and be able to deploy your HCP Consul using
+Terraform, you have to create a Service Principal and a key associated to it.
+
+From the left menu select _Access Control (IAM)_.
+On the left menu of the _Access Control (IAM)_ page, click on the _Service Principals_ tab, and on
+the page click on the create link.
+
+Use the name you prefer to create the Service Principal (we used `xx-2048` for
+this use case) but remember to use `Contributor` as the role.
+
+Once the Service Principal is created, click on the service principal name on
+the page to view its details. From the detail page, click on **+ Generate key**.
+
+~> **Note:** Remember to copy the Client ID and secret; you won't be able to
+retrieve the secret later once you close the pop up.
+
+Save the client ID and secret as the environment variables `HCP_CLIENT_ID` and `HCP_CLIENT_SECRET`.
+Or, configure the provider with the client ID and secret by copy-pasting the values directly into provider config.
+
+```terraform
+// Credentials can be set explicitly or via the environment variables HCP_CLIENT_ID and HCP_CLIENT_SECRET
+provider "hcp" {
+client_id     = "service-principal-key-client-id"
+client_secret = "service-principal-key-client-secret"
+}
+```
+
+```shell
+export HCP_CLIENT_ID="service-principal-key-client-id"
+export HCP_CLIENT_SECRET="service-principal-key-client-secret"
+```
+
+You can view existing keys and generate new ones on this page:
+![HCP Service Principal](/img/use-cases/hcp-hashistack-quickstart/service-principal.png)
+
+When client credentials are set, they are always used by the HCP Provider client, regardless of an existing user session.
+
+## Define the Virtual Network
+
+### VPC
+
+You can find VPC configurations in `vpc.tf`. There is more than one way to create this VPC, including manually in AWS.
+The Amazon VPC can be managed with the [AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs).
+
+```terraform
+provider "aws" {
+   region = var.region
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+  name   = var.name
+  cidr   = var.cidr
+  public_subnets = var.public_subnets
+  public_subnet_tags = {
+    Name = "${var.name}-public"
+  }
+
+  azs            = slice(data.aws_availability_zones.available.*.names[0], 0, 3)
+   enable_dns_hostnames            = true
+  enable_ipv6                     = true
+  assign_ipv6_address_on_creation = true
+  public_subnet_ipv6_prefixes     = [0, 1, 2]
+
+  tags = {
+    Terraform = "true"
+    Name      = var.name
+  }
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+```
+
+### HVN
+
+In the [`hcp.tf`](https://github.com/xiaolin-ninja/2048_hcp_hashistack/blob/main/hcp.tf) configuration file,
```

Comment:
For links like this, I wonder if it makes more sense to link directly to the latest commit sha rather than `main`? If that file ever gets moved, we'll have dead links in the guide. Linking to a specific commit in the source tree would at least keep it active.